code/game/machinery/camera/tracking.dm

```diff
@@ -183,34 +181,48 @@
 
 	return TRUE
 
-/mob/living/silicon/ai/proc/ai_actual_track(var/atom/target)
-	if(!istype(target))
+/mob/living/silicon/ai/proc/ai_actual_track(var/atom/movable/target)
+	if(!istype(target) || !eyeobj)
 		return
 
 	if(!can_track_atom(target))
-		to_chat(src, "Target is not near any active camera.")
+		to_chat(src, "Target is not trackable by any means.")
 		return
 
-	cameraFollow = target
-
-	to_chat(src, "Now tracking [target.name] on camera.")
-
-	spawn (0)
-		while (cameraFollow == target)
-			if (cameraFollow == null)
-				return
-
-			if(!can_track_atom(target))
-				to_chat(src, "Target is not near any active camera.")
-				sleep(10 SECONDS)
-				continue
-
-			if(eyeobj)
-				eyeobj.forceMove(get_turf(target))
-			else
-				view_core()
-				return
-			sleep(1 SECONDS)
+	stop_ai_tracking()
+	currently_tracking = target
+	eyeobj.forceMove(get_turf(currently_tracking))
+	currently_tracking.lock_atom(eyeobj,/datum/locking_category/ai_eye)
+
+	currently_tracking.register_event(/event/moved,src,nameof(src::on_camera_change()))
+	currently_tracking.register_event(/event/equipped,src,nameof(src::on_camera_change()))
+	//currently_tracking.register_event(/event/unequipped,src,nameof(src::on_camera_change()))
+	currently_tracking.register_event(/event/cameranet_changed,src,nameof(src::on_camera_change()))
+	to_chat(src, "Now tracking [currently_tracking.name] on camera.")
+
+/datum/locking_category/ai_eye
+
+/mob/living/silicon/ai/proc/on_camera_change(mover)
```

Comment:
Is the mover variable used at any point?

Reply:
nah, just needed so the event can be invoked